### PR TITLE
libndctl: 71.1 -> 79

### PR DESCRIPTION
**Pull request by Kaisrlik**
Bumping and modifying build according to meson build system. Autotools were removed in version 73.

## Description of changes

* Version was bumped due to missing `cxl` binary that was added in version `72`.
* other changes https://github.com/pmem/ndctl/tags
* my first PR here, please let me know if I'm missing anything :see_no_evil: 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by smancill** [comment]
Please ask in https://discourse.nixos.org/t/prs-ready-for-review/ for a review.

Looks OK to me but I cannot build it since I only have a MacBook.

File: pkgs/development/libraries/libndctl/default.nix [hunk 1]
```diff
@@ -1,43 +1,65 @@
...
+  mesonFlags =
+    [
+      "-Drootprefix=${placeholder "out"}"
+      "-Dsysconfdir=${placeholder "out"}/etc/ndctl.conf.d"
+      "-Dlibtracefs=disabled"
+      # Use asciidoctor due to xmlto errors
+      "-Dasciidoctor=enabled"
+      "-Dsystemd=disabled"
+      "-Diniparserdir=${iniparser}"
     ];
```
Comment (eljamm) [inline pkgs/development/libraries/libndctl/default.nix:56]: This looks much clearer to me:

```suggestion
  mesonFlags = [
    (lib.mesonOption "rootprefix" "${placeholder "out"}")
    (lib.mesonOption "sysconfdir" "${placeholder "out"}/etc/ndctl.conf.d")
    (lib.mesonEnable "libtracefs" false)
    # Use asciidoctor due to xmlto errors
    (lib.mesonEnable "asciidoctor" true)
    (lib.mesonEnable "systemd" false)
    (lib.mesonOption "iniparserdir" "${iniparser}")
  ];
```

File: pkgs/development/libraries/libndctl/default.nix [hunk 2]
```diff
@@ -1,43 +1,65 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook
-, asciidoc, pkg-config, xmlto, docbook_xsl, docbook_xml_dtd_45, libxslt
-, json_c, kmod, which, util-linux, udev, keyutils
+{ lib
```
Comment (eljamm) [inline pkgs/development/libraries/libndctl/default.nix:1]: You need to format the file with [nixfmt](https://github.com/NixOS/nixfmt)

File: pkgs/development/libraries/libndctl/default.nix [hunk 3]
```diff
@@ -1,43 +1,65 @@
...
+      "-Diniparserdir=${iniparser}"
     ];
 
   patchPhase = ''
     patchShebangs test
 
     substituteInPlace git-version --replace /bin/bash ${stdenv.shell}
     substituteInPlace git-version-gen --replace /bin/sh ${stdenv.shell}
-
-    echo "m4_define([GIT_VERSION], [${version}])" > version.m4;
```
Comment (eljamm) [inline pkgs/development/libraries/libndctl/default.nix:40]: Just curious, why was this removed?

File: pkgs/development/libraries/libndctl/default.nix [hunk 4]
```diff
@@ -1,43 +1,65 @@
...
+      "-Drootprefix=${placeholder "out"}"
+      "-Dsysconfdir=${placeholder "out"}/etc/ndctl.conf.d"
+      "-Dlibtracefs=disabled"
+      # Use asciidoctor due to xmlto errors
+      "-Dasciidoctor=enabled"
+      "-Dsystemd=disabled"
+      "-Diniparserdir=${iniparser}"
     ];
 
   patchPhase = ''
```
Comment (eljamm) [inline pkgs/development/libraries/libndctl/default.nix:58]: When a phase is overriden, you need to add its `pre-` and `post-` hooks:

```suggestion
  patchPhase = ''
    runHook prePatch

    patchShebangs test

    substituteInPlace git-version --replace /bin/bash ${stdenv.shell}
    substituteInPlace git-version-gen --replace /bin/sh ${stdenv.shell}

    runHook postPatch
  '';
```

File: pkgs/development/libraries/libndctl/default.nix (hunk 3 above)
Comment (eljamm) [reply pkgs/development/libraries/libndctl/default.nix:40]: Thanks. If the output is the same then it's alright. I'll leave the decision to keep or remove it up to you.

File: pkgs/development/libraries/libndctl/default.nix [hunk 5]
```diff
@@ -1,50 +1,50 @@
...
-    echo "m4_define([GIT_VERSION], [${version}])" > version.m4;
+    runHook postPatch
   '';
 
   meta = with lib; {
-    description = "Tools for managing the Linux Non-Volatile Memory Device sub-system";
-    homepage    = "https://github.com/pmem/ndctl";
-    license     = licenses.lgpl21;
+    description =
+      "Tools for managing the Linux Non-Volatile Memory Device sub-system";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libndctl/default.nix:44]:
```suggestion
    description = "Tools for managing the Linux Non-Volatile Memory Device sub-system";
```

File: pkgs/development/libraries/libndctl/default.nix [hunk 6]
```diff
@@ -1,50 +1,50 @@
...
+    (lib.mesonOption "rootprefix" "${placeholder "out"}")
+    (lib.mesonOption "sysconfdir" "${placeholder "out"}/etc/ndctl.conf.d")
+    (lib.mesonEnable "libtracefs" false)
+    # Use asciidoctor due to xmlto errors
+    (lib.mesonEnable "asciidoctor" true)
+    (lib.mesonEnable "systemd" false)
+    (lib.mesonOption "iniparserdir" "${iniparser}")
+  ];
 
   patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libndctl/default.nix:31]: You should never ever overwrite patchPhase as that breaks patches. Please move all of this to postPatch

File: pkgs/development/libraries/libndctl/default.nix [hunk 7]
```diff
@@ -1,50 +1,50 @@
...
+    runHook prePatch
+
     patchShebangs test
 
     substituteInPlace git-version --replace /bin/bash ${stdenv.shell}
     substituteInPlace git-version-gen --replace /bin/sh ${stdenv.shell}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libndctl/default.nix:37]:
```suggestion
    substituteInPlace git-version --replace-fail /bin/bash ${stdenv.shell}
    substituteInPlace git-version-gen --replace-fail /bin/sh ${stdenv.shell}
```

File: pkgs/development/libraries/libndctl/default.nix [hunk 8]
```diff
@@ -1,50 +1,50 @@
...
-      "--disable-asciidoctor" # depends on ruby 2.7, use asciidoc instead
-    ];
+  mesonFlags = [
+    (lib.mesonOption "rootprefix" "${placeholder "out"}")
+    (lib.mesonOption "sysconfdir" "${placeholder "out"}/etc/ndctl.conf.d")
+    (lib.mesonEnable "libtracefs" false)
+    # Use asciidoctor due to xmlto errors
+    (lib.mesonEnable "asciidoctor" true)
+    (lib.mesonEnable "systemd" false)
+    (lib.mesonOption "iniparserdir" "${iniparser}")
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libndctl/default.nix:28]:
```suggestion
    (lib.mesonOption "iniparserdir" iniparser)
```

File: pkgs/development/libraries/libndctl/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/libndctl/default.nix:40]: If the version ends up in the application correctly, it doesn't matter how we got it in. Can we set the new option?

File: pkgs/development/libraries/libndctl/default.nix [hunk 9]
```diff
@@ -1,50 +1,50 @@
...
-  nativeBuildInputs =
-    [ autoreconfHook asciidoc pkg-config xmlto docbook_xml_dtd_45 docbook_xsl libxslt
-      which
-    ];
+  nativeBuildInputs = [ meson ninja pkg-config ];
 
-  buildInputs =
-    [ json_c kmod util-linux udev keyutils
-    ];
+  buildInputs = [ asciidoctor iniparser json_c keyutils kmod udev util-linux ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libndctl/default.nix:19]: Isn't asciidoctor a cli?

File: pkgs/development/libraries/libndctl/default.nix (hunk 3 above)
Comment (Kaisrlik) [reply pkgs/development/libraries/libndctl/default.nix:40]: The version is already part of the file - https://github.com/pmem/ndctl/blob/16f45755f991f4fb6d76fec70a42992426c84234/git-version#L22 . In case we want to apply some suffix it want be propagated to build, so let me revert it to original approach.

File: pkgs/development/libraries/libndctl/default.nix (hunk 9 above)
Comment (Kaisrlik) [reply pkgs/development/libraries/libndctl/default.nix:19]: yes, it seem it has cli api - https://asciidoctor.org/#command-line-interface-cli

File: pkgs/development/libraries/libndctl/default.nix (hunk 8 above)
Comment (Kaisrlik) [reply pkgs/development/libraries/libndctl/default.nix:28]: I guess, this is not correct(and does not work for me). I need to pass a string and the string needs to point to location of iniparser dir.

File: pkgs/development/libraries/libndctl/default.nix (hunk 5 above)
Comment (Kaisrlik) [reply pkgs/development/libraries/libndctl/default.nix:44]: Nixfmt is keep falling back to original format, should I follow your advice or format user by nixfmt?

Comment (eljamm) [reply pkgs/development/libraries/libndctl/default.nix:42]: Are you using the wrong nixfmt by any chance? You should use [nixfmt-rfc-style](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=nixfmt-rfc-style).

Comment (Kaisrlik) [reply pkgs/development/libraries/libndctl/default.nix]: Thanks! Yes, that was the problem. Sorry.

**Review by eljamm — APPROVED** [review]
This looks good to me :+1:

**Review by smancill — APPROVED** [review]
